srv/SpawnEntity.srv

```diff
@@ -0,0 +1,16 @@
+# Spawn an entity (a robot, other object) by name or URI
+
+string name                         # Suggest a name to give to the spawned entity. If left empty, it will be generated.
+                                    # Due to requirement for name uniqueness and the namespace, check the status_message
+                                    # to get an actual unique name to use for further queries.
+string uri                          # Resource such as SDF or URDF file, or in native format such as prefab.
+                                    # Valid URIs can be determined by calling GetSpawnables first,
+                                    # if the feature is supported by your simulator.
+string robot_namespace              # Spawn robot or another object with all its interfaces under this namespace.
+string reference_frame              # Optional initial pose reference frame. Defaults to global "world" frame.
```

Comment:
What values are allowed in here? Is the concept of frame defined somewhere?

Reply:
Yes, it is a ROS frame, same semantics as anywhere else. I added a clarification:

This ROS frame must be known to the simulator, e.g. of an object spawned earlier.

It makes sense to spawn an obstacle in the reference frame of the robot you spawned earlier, e.g. a meter in front of the robot. SpawnEntity does not need to make use of predermined spawn points, these are for convenience.

Reply:
Wait wait wait. ROS frame? I thought each simulator has its own set of frames. To get ROS frames out of them, there usually needs to be a bridge translating them to TF messages, and possibly translating/renaming some of them. And I don't know how I would get a TF frame that "originates" in ROS (i.e. a node computes and publishes it) into the simulator (what would the frame be? an Entity?).

Reply:
Yes, but given it is a ROS standard for simulation, I would expect implementations to handle the mapping, and respond with error if they cannot identify the frame. Simulators can simply support empty / default mode as well.

Reply:
Okay, it would be interesting to have an opinion on this from some of the Gazebo authors.